### PR TITLE
Missing element titles

### PR DESCRIPTION
**Pull request by bxo11**
PR introduces the titles to icon elements without text value.

### CODE REVIEW

File: indico/modules/attachments/templates/_attachments.html [hunk 1]
```diff
@@ -41,11 +41,18 @@
...
                     {% else %}
                         {% trans name=attachment.title %}Edit link &quot;{{ name }}&quot;{% endtrans %}
-                    {% endif -%}"></a>
+                    {% endif -%}"
+                    title="
+                    {%- if attachment.type.name == 'file' -%}
+                        {% trans name=attachment.title %}Edit file &quot;{{ name }}&quot;{% endtrans %}
+                    {% else %}
+                        {% trans name=attachment.title %}Edit link &quot;{{ name }}&quot;{% endtrans %}
+                    {% endif -%}"></a>></a>
```
Comment (ThiefMaster) [inline indico/modules/attachments/templates/_attachments.html:50]: that looks wrong...

Comment (ThiefMaster) [reply indico/modules/attachments/templates/_attachments.html:50]: BTW if you want to do something more fancy that avoids duplication, you could rename `data-title` to `title` and add `data-title` without a value, and adapt the JS that opens these dialogs to use the `title` if `data-title` is present (not undefined) but empty

File: indico/modules/users/templates/dashboard.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
 {%- endblock %}
 
 {% block banner_actions -%}
-    <div id="dashboard-calendar-link"></div>
+    <div id="dashboard-calendar-link" title="{% trans %}Export{% endtrans %}"></div>
```
Comment (ThiefMaster) [inline indico/modules/users/templates/dashboard.html:31]: Does this still stay when the `ICSCalendarLink` component gets rendered into this container? I thought the container gets replaced but maybe it's just its contents...

File: indico/modules/events/layout/templates/_menu.html [hunk 3]
```diff
@@ -22,17 +22,19 @@
...
                 {% if not entry.is_separator %}
-                    <a href="#" data-href="{{ url_for('.menu_entry_edit', entry) }}" class="edit-entry"></a>
+                    <a href="#" title="{% trans%}Edit entry{% endtrans %}" data-href="{{ url_for('.menu_entry_edit', entry) }}" class="edit-entry"></a>
                 {% endif %}
                 {% if entry.type.name in ('user_link', 'page', 'separator') %}
                     <a href="#" class="delete-entry"
                          data-method="DELETE"
                          data-href="{{ url_for('.menu_delete_entry', entry) }}"
                          data-title="{% trans%}Remove the menu entry?{% endtrans %}"
+                         title="{% trans%}Remove the menu entry{% endtrans %}"
```
Comment (ThiefMaster) [inline indico/modules/events/layout/templates/_menu.html:37]:
```suggestion
                         data-title="{% trans %}Remove the menu entry?{% endtrans %}"
                         title="{% trans %}Remove the menu entry{% endtrans %}"
```

File: indico/modules/events/layout/templates/_menu.html [hunk 4]
```diff
@@ -22,17 +22,19 @@
                     ></a>
                 {% endif %}
                 <a class="{% if not entry.is_enabled %}not-{% endif %}enabled" href="#"
+                   title="{% trans %}Toggle visibility{% endtrans %}"
                    data-href="{{ url_for('.menu_entry_toggle_enabled', entry) }}"
                    data-method="POST"
                 ></a>
                 {% if not entry.is_separator %}
-                    <a href="#" data-href="{{ url_for('.menu_entry_edit', entry) }}" class="edit-entry"></a>
+                    <a href="#" title="{% trans%}Edit entry{% endtrans %}" data-href="{{ url_for('.menu_entry_edit', entry) }}" class="edit-entry"></a>
```
Comment (ThiefMaster) [inline indico/modules/events/layout/templates/_menu.html:30]:
```suggestion
                    <a href="#" title="{% trans %}Edit entry{% endtrans %}" data-href="{{ url_for('.menu_entry_edit', entry) }}" class="edit-entry"></a>
```

File: indico/modules/users/templates/dashboard.html (hunk 2 above)
Comment (ThiefMaster) [reply indico/modules/users/templates/dashboard.html:31]: FWIW if you include the title inside the react component you can omit it in all the places where it gets rendered (e.g. `indico/modules/categories/templates/display/base.html` as well)

**Comment by ThiefMaster** [comment]
I think it'd still be nice to consider my suggestion from [this comment](https://github.com/indico/indico/pull/5450#discussion_r944571550) a try to avoid duplicate all those titles in the template. I tried this and it seems to work fine (maybe there are other places opening dialogs that need this logic, not sure):

```diff
diff --git a/indico/modules/attachments/client/js/legacy.js b/indico/modules/attachments/client/js/legacy.js
index 216b93390a..76f928cb9f 100644
--- a/indico/modules/attachments/client/js/legacy.js
+++ b/indico/modules/attachments/client/js/legacy.js
@@ -184,10 +184,15 @@
       .on('click', '.js-dialog-action', function(e) {
         e.preventDefault();
         const $this = $(this);
+        let title = $this.data('title');
+        if (!title && title !== undefined) {
+          // if `data-title` is present without a value, fall back to the title attr
+          title = $this.attr('title') || $this.data('qtip-oldtitle');
+        }
         ajaxDialog({
           trigger: this,
           url: $this.data('href'),
-          title: $this.data('title'),
+          title,
           hidePageHeader: true,
           onClose(data) {
             if (data) {
diff --git a/indico/modules/attachments/templates/_attachments.html b/indico/modules/attachments/templates/_attachments.html
index 9fa76d0fbf..5fed3dba61 100644
--- a/indico/modules/attachments/templates/_attachments.html
+++ b/indico/modules/attachments/templates/_attachments.html
@@ -79,7 +79,8 @@
             <td class="actions hide-if-locked">
                 <a class="icon-edit js-dialog-action"
                            data-href="{{ url_for('attachments.edit_folder', folder) }}"
-                           data-title="{% trans name=folder.title %}Edit folder: {{ name }}{% endtrans %}"></a>
+                           data-title
+                           title="{% trans name=folder.title %}Edit folder: {{ name }}{% endtrans %}"></a>
                 <a href="#" class="icon-remove js-delete"
                    data-method="DELETE"
                    data-href="{{ url_for('attachments.delete_folder', folder) }}"
diff --git a/indico/web/client/js/jquery/utils/declarative.js b/indico/web/client/js/jquery/utils/declarative.js
index 4ab149d948..c66efb7b8d 100644
--- a/indico/web/client/js/jquery/utils/declarative.js
+++ b/indico/web/client/js/jquery/utils/declarative.js
@@ -151,13 +151,18 @@ import {$T} from '../../utils/i18n';
         // Handle AJAX dialog
         if (dialog) {
           var closeButton = $this.data('close-button');
+          let title = $this.data('title');
+          if (!title && title !== undefined) {
+            // if `data-title` is present without a value, fall back to the title attr
+            title = $this.attr('title') || $this.data('qtip-oldtitle');
+          }
           ajaxDialog({
             trigger: $this,
             url: url,
             method: method,
             data: params,
             content: $(content).html(),
-            title: $this.data('title'),
+            title,
             subtitle: $this.data('subtitle'),
             closeButton: closeButton === undefined ? false : closeButton || true,
             dialogClasses: $this.data('dialog-classes'),
```

**Comment by bxo11** [comment]
Thanks for the suggestion, I applied it where `data-title` and `title` are the same

File: indico/modules/users/templates/dashboard.html (hunk 2 above)
Comment (bxo11) [reply indico/modules/users/templates/dashboard.html]: > Does this still stay when the `ICSCalendarLink` component gets rendered into this container? I thought the container gets replaced but maybe it's just its contents...

Yes, it remains the same

File: indico/modules/attachments/client/js/legacy.js [hunk 5]
```diff
@@ -184,10 +184,15 @@
       .on('click', '.js-dialog-action', function(e) {
         e.preventDefault();
         const $this = $(this);
+        let title = $this.data('title');
+        if (!title && title !== undefined) {
+         // if `data-title` is present without a value, fall back to the title attr
+         title = $this.attr('title') || $this.data('qtip-oldtitle');
```
Comment (ThiefMaster) [inline indico/modules/attachments/client/js/legacy.js:190]:
```suggestion
          // if `data-title` is present without a value, fall back to the title attr
          title = $this.attr('title') || $this.data('qtip-oldtitle');
```

This should be using 2-space indentation!

File: indico/modules/events/templates/display/conference.html [hunk 6]
```diff
@@ -90,6 +90,7 @@
                     {% if is_submitter %}
                         <div>
                             <a href="#" class="i-button text-color subtle icon-edit hide-if-locked"
+                               title="{% trans %}Manage material{% endtrans %}"
```
Comment (tomasr8) [inline indico/modules/events/templates/display/conference.html:93]:
```suggestion
                               title="{% trans %}Manage materials{% endtrans %}"
```

**Comment by tomasr8** [comment]
Hey, thanks for the PR! I think the pdf and calendar export buttons on session and contribution pages could also use titles:
image

**Review by tomasr8 — APPROVED** [review]

File: indico/web/client/js/jquery/widgets/jinja/permissions_widget.js [hunk 7]
```diff
@@ -193,6 +194,7 @@ import Palette from 'indico/utils/palette';
       const self = this;
       return $('<button>', {
         'type': 'button',
+        'title': $T.gettext("Delete entry '{0}'".format(principal.name || principal.id)),
```
Comment (ThiefMaster) [inline indico/web/client/js/jquery/widgets/jinja/permissions_widget.js:197]: Formatting needs to happen outside the gettext call:

```suggestion
        'title': $T.gettext("Delete entry '{0}'").format(principal.name || principal.id),
```

**Review by ThiefMaster — CHANGES_REQUESTED** [review]

**Review by ThiefMaster — APPROVED** [review]

**Comment by ThiefMaster** [comment]
Thanks!

**Pull request merged**